ROX13642 Creating Collector Troubleshooting guide (#54763)
openshift/openshift-docs: troubleshooting/commonly-occurring-error-conditions.adoc

:_content-type: ASSEMBLY
[id="commonly-occurring-error-conditions"]
= Commonly occurring error conditions
include::modules/common-attributes.adoc[]
:context: error-conditions

toc::[]

[role="_abstract"]

> The vast majority of errors occur during Collector startup, when the Collector configures itself, finds or downloads an appropriate kernel driver for the system, and then starts collecting events. 

Reviewer: ```suggestion
Most errors occur during Collector startup when Collector configures itself and finds or downloads a kernel driver for the system. 
```

I am not sure about `and then starts collecting events.` does error occur during this phase?

Reviewer 2: Errors that occur after Collector has begun collecting events are more difficult to generally troubleshoot (and are somewhat outside of the scope of these docs), so I agree that this should be removed from this paragraph.

Author: ack

The following diagram describes the main parts of the startup process:

> <Diagram>

Reviewer: Since we don't have a diagram, we can remove this part. We can revisit the content again when we have the diagram.

Author: ack

> If any part of the start-up procedure fails, a diagnostic summary is written to the logs, detailing which steps succeeded or failed. 

Author: ack

> The following log file example shows a successful startup:  

Author: ack

[source,terminal]
----
[INFO    2022/11/28 13:21:55] == Collector Startup Diagnostics: ==
[INFO    2022/11/28 13:21:55]  Connected to Sensor?       true
[INFO    2022/11/28 13:21:55]  Kernel driver available?   true
[INFO    2022/11/28 13:21:55]  Driver loaded into kernel? true
[INFO    2022/11/28 13:21:55] ====================================
----

> It confirms that Collector connected to Sensor and located and loaded the kernel driver. 

Reviewer: Unclear antecedent `It`

Author: ack
